app/forms/state_file/nj_homeowner_eligibility_form.rb

```diff
@@ -0,0 +1,21 @@
+module StateFile
+  class NjHomeownerEligibilityForm < QuestionsForm
+    set_attributes_for :intake,
+                       :homeowner_home_subject_to_property_taxes,
+                       :homeowner_main_home_multi_unit,
+                       :homeowner_main_home_multi_unit_max_four_one_commercial,
+                       :homeowner_more_than_one_main_home_in_nj,
+                       :homeowner_shared_ownership_not_spouse
+
+    def save
+      is_hard_no = (self.homeowner_home_subject_to_property_taxes == 'no' or (self.homeowner_main_home_multi_unit == 'yes' and self.homeowner_main_home_multi_unit_max_four_one_commercial == 'no'))
+      is_soft_no = (self.homeowner_more_than_one_main_home_in_nj == 'yes' or self.homeowner_shared_ownership_not_spouse == 'yes' or (self.homeowner_main_home_multi_unit == 'yes' and self.homeowner_main_home_multi_unit_max_four_one_commercial == 'yes'))
+
+      if is_soft_no || is_hard_no
+        @intake.property_tax_paid = nil
```

Comment:
should this be `@intake.update(property_tax_paid: nil)`? it looks like this will currently be saved by the `@intake.update` call at the end of this method, but i wouldn't want to rely on the way the code is currently ordered.

Reply:
@mrotondo This is a useful callout, is your suggestion the preferred approach? I was borrowing from the approach in other conditional screens where changing one answer would reset the following dependent screen - specifically was seeing it in `NyCountyForm`. Happy to do either approach, what is the preference here?

Reply:
I think I may have miscommunicated - I think this method is well structured & doing all the right things, I was just noting that it combines assignment (`@intake.property_tax_paid = nil`, which doesn't result in anything being saved to the database) with `update` (`@intake.update(attributes_for(:intake))`, which calls `save` behind the scenes, and so ensures that the previous assignment will get persisted).

It's a minor stylistic concern, and one you should feel free to ignore, as the code will currently do exactly what you meant for it to do - should have marked it [DUST], I think is the right word?

I'd prefer either
1. multiple assigns followed by an explicit `save` call (as `NyCountyForm` does, which ensures that the whole method's changes go in a single database transaction), or
2. no assigns, but multiple `updates` and no `save` call, which doesn't rely on the order of lines staying fixed.